contrib/auto_explain/auto_explain.c

```diff
@@ -52,7 +52,8 @@ static ExecutorEnd_hook_type prev_ExecutorEnd = NULL;
 
 #define auto_explain_enabled() \
 	(auto_explain_log_min_duration >= 0 && \
-	 (nesting_level == 0 || auto_explain_log_nested_statements))
+	 (nesting_level == 0 || auto_explain_log_nested_statements) && \
+	 (Gp_role == GP_ROLE_DISPATCH))
```

Comment:
I'd like to not disable auto_explain for utility mode then conjunct `Gp_role != GP_ROLE_EXECUTE` looks more reasonable.

Reply:
I don't know, how auto_explain will work at utility mode. At _PG_init there is comment^
`/* Only run auto_explain on the Query Dispatcher node */`

Reply:
I propose to look at [PR](https://github.com/greenplum-db/gpdb/pull/6100) (and parent [issue](https://github.com/greenplum-db/gpdb/issues/5906)) where this guard check was introduced. `auto_explain` really has the logic regarding to work in cluster but in utility mode this might not be critical. I propose to manually check the work of extension without guard check in `PG_init()` in utility mode.